src/elementDisabler.ts

```diff
@@ -80,6 +80,12 @@ function disableFormElements (form: HTMLFormElement): void {
 function disableFormElement (element: HTMLFormElement): void {
   if (element.dataset.ujsDisabled != null) return
 
+  const form = element?.form
+
+  // If the form is using the browser's default constraint validation, and the form is not valid, we should
+  // not attempt to disable the button because the browser will prevent errors and prevent submission
```

Comment:
Wonder if we can boil the comment down a bit? Here's a pitch:

```suggestion
  // When browser constraint validation using forms are invalid, the browser prevents errors & submission
```